### PR TITLE
Introduce an inherited widget to provide controllers that live longer than an individual page.

### PR DESCRIPTION
**Pull request by DaveShuckerow**
This satisfies one of the requirements of #1286.  We need the logging controller active even when we're not on the logging page.

### CODE REVIEW

File: packages/devtools_app/lib/src/flutter/provider.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import 'package:flutter/material.dart';
+
+import '../auto_dispose.dart';
+import '../globals.dart';
+import '../logging/logging_controller.dart';
+
+/// Container for controllers that should outlive individual screens of the app.
+@immutable
+class ProviderData implements DisposableController {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/flutter/provider.dart]: if this class is intended to be just for controllers, perhaps rename to ControllerProviderData. Same for below: r/Provider/ControllerProvider

File: packages/devtools_app/lib/src/flutter/provider.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+/// Provider for controllers that should outlive individual screens of the app.
+///
+/// [Initializer] builds a [Provider] after it has a connection to the VM
+/// service and it has loaded [ensureInspectorDependencies].
+class Provider extends StatefulWidget {
+  const Provider({Key key, this.child, this.overrideProviders})
+      : super(key: key);
+
+  final ProviderData Function() overrideProviders;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/flutter/provider.dart]: nit: can this method be `@visibleForTesting` ?

File: packages/devtools_app/test/flutter/initializer_test.dart [hunk 3]
```diff
@@ -2,18 +2,26 @@
...
 import 'package:devtools_app/src/globals.dart';
 import 'package:devtools_app/src/service_manager.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
 
+import '../support/mocks.dart';
+
 void main() {
   group('Initializer', () {
+    FakeServiceManager serviceManager;
```
Comment (jacob314) [inline packages/devtools_app/test/flutter/initializer_test.dart]: is this variable used? If not maybe revert this change and set it directly in the setGlobal call like it was before.

Comment (DaveShuckerow) [reply packages/devtools_app/test/flutter/initializer_test.dart]: No; done.

File: packages/devtools_app/lib/src/flutter/provider.dart (hunk 1 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/flutter/provider.dart]: That's a good point, done.

We'll need to talk to these classes in a lot of places, so I've decided to go with a slightly shorter name.

We'll have `Controllers.of(context).logging` to get the logging controller, for instance.

File: packages/devtools_app/lib/src/flutter/provider.dart (hunk 2 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/flutter/provider.dart]: Done.

File: packages/devtools_app/lib/src/flutter/provider.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/flutter/provider.dart]: +100 on shorter names.
